marshmallow/fields.py

```diff
@@ -386,6 +386,22 @@ def __init__(self, nested, default=missing_, exclude=tuple(), only=None, **kwarg
         self.many = kwargs.get('many', False)
         self.__schema = None  # Cached Schema instance
         self.__updated_fields = False
+
+        if not isinstance(self.many, bool):
+            raise ValueError('Many parameter must be boolean')
```

Comment:
`many` does not have to be a boolean. You can pass anything, as long as it evaluates to a either truthy value if you expect a collection or a falsy value if you expect a single element.